### PR TITLE
Prefer site.tagline to site.description for page title

### PR DESCRIPTION
**Pull request by qwtel**
Using `site.description` as part of the title for pages that don't have an explicit title --- typically the landing page of a site --- leads to very long page titles. The `site.description` field is typically used for search engine or social media previews, which are 100+ characters long, making it a bad choice for the title for pages that use the description field in this way.

This PR changes the behavior of this plugin to prefer `site.tagline` over `site.description` when it is available. Sites that do not define a `site.tagline` are not affected by this. 

The name `site.tagline` was chosen as it served [the same role][1] in the Hyde theme. 

[1]: https://github.com/poole/hyde/blob/master/_includes/head.html

### CODE REVIEW

**Comment by ashmaroli** [comment]
@qwtel Can you add a test for a scenario that doesn't have `site.tagline` defined? Thanks.

**Comment by qwtel** [comment]
I didn't add one specifically b/c `site.tagline` being undefined is the case for the entire current tests suit.

**Comment by qwtel** [comment]
Can this be merged now?

File: lib/jekyll-seo-tag/drop.rb [hunk 1]
```diff
@@ -43,14 +47,18 @@ def page_title
...
+
       # Page title with site title or description appended
       # rubocop:disable Metrics/CyclomaticComplexity
       def title
         @title ||= begin
           if site_title && page_title != site_title
             page_title + TITLE_SEPARATOR + site_title
           elsif site_description && site_title
-            site_title + TITLE_SEPARATOR + site_description
+            site_title + TITLE_SEPARATOR + site_tagline_or_description
```
Comment (parkr) [inline lib/jekyll-seo-tag/drop.rb]: If this is the only place `site_tagline_or_description` is used, I’d suggest inlining the method. Feel free to use parentheses.

Comment (qwtel) [reply lib/jekyll-seo-tag/drop.rb]: I tried to do that, but it created a style offence: 
* `Metrics/AbcSize: Assignment Branch Condition size for title is too high`
* `Metrics/PerceivedComplexity: Perceived complexity for title is too high`

**Comment by qwtel** [comment]
Will update with documentation

**Comment by qwtel** [comment]
I've updated the documentation, how does it look?

**Comment by qwtel** [comment]
Can this be merged now? 😉

**Comment by DirtyF** [comment]
@jekyllbot: merge +minor

**Pull request merged**